test/go-tests/test_utils.go

```diff
@@ -1141,3 +1145,105 @@ func resetTestPath(t *testing.T, path string) {
 	err := os.Chdir(path)
 	require.Nil(t, err)
 }
+
+func recursiveZip(source, target string, needBaseDir bool) error {
```

Comment:
I think that the name `recursiveZip` is a little bit misleading for this function; I would suggest something like `createZipFileFromDirectory` could make the code more readable

Reply:
Just renamed it